templates/lib/yarp/node.rb.erb

```diff
@@ -48,6 +48,31 @@ module YARP
       }.compact.join(", ") %>]
     end
 
+    # def compact_child_nodes: () -> Array[Node]
+    def compact_child_nodes
+      <%- if node.fields.any? { |field| field.is_a?(YARP::OptionalNodeField) } -%>
```

Comment:
Another possibility would be to use `[node_field, *optional_node_field, *node_list_field]` because there is no `#to_a` on Node and `*nil` is `[]`.
Not sure if faster or slower though, it would make the logic here a bit simpler.